rke/structure_rke_cluster.go

```diff
@@ -505,3 +516,26 @@ func expandRKEClusterFlag(in *schema.ResourceData, clusterFilePath string) clust
 
 	return obj
 }
+
+func k8sVersionRequiresCri(kubernetesVersion string) bool {
```

Comment:
I got the idea here, but there is an easier way to achieve this.  
Please check out [this function](https://github.com/rancher/rke/blob/645e80840596a8616c2a1f35cea0c655ac700fa9/cluster/defaults.go#L332-L346) where RKE checks and sets the enable_cri_dockerd when certain conditions are met.